### PR TITLE
Adding Pagination to get_gas_objects

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Adding Pagination to get_gas_objects
## Test Plan 

CI
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui/src/client_commands.rs [hunk 1]
```diff
@@ -1263,32 +1263,46 @@ impl WalletContext {
...
+                            Some(SuiObjectDataOptions::full_content()),
+                        )),
+                        response.next_cursor,
+                        None,
+                    )
+                    .await?
+            } else {
+                has_next_page = false;
+            }
+        }
```
Comment (amnn) [inline crates/sui/src/client_commands.rs:1300]: Could maybe simplify this a little:

```rust
let mut cursor = None;
loop {
    let response = client
        .read_api()
        .get_owned_objects(
            address,
            Some(SuiObjectResponseQuery::new(
                Some(SuiObjectDataFilter::StructType(GasCoin::type_()),
                Some(SuiObjectDataOptions::full_content()),
            )),
            cursor,
            None,
        )
        .await?;

    objects.extend(response.data);
    
    if response.has_next_page {
        cursor = response.next_cursor;
    } else {
        break
    }
}
```

**Review by amnn — COMMENTED** [review]
I'll leave @666lcz to give the final go ahead, but this change looks great, thanks @healthydeve !  Just left a comment for a potential simplification.

**Review by 666lcz — CHANGES_REQUESTED** [review]

File: crates/sui/src/client_commands.rs [hunk 2]
```diff
@@ -1263,32 +1263,46 @@ impl WalletContext {
...
-        // TODO: We should ideally fetch the objects from local cache
-        let mut values_objects = Vec::new();
+            .await?;
 
-        let oref_ids = objects
-            .into_iter()
-            .map(|o| o.object().unwrap().object_id)
-            .collect();
+        while has_next_page {
+            objects.extend(response.data.clone());
```
[the remark this answers is not in the document]
Comment (healthydeve) [reply crates/sui/src/client_commands.rs:1283]: Fixed with the loop ashok suggested

**Review by 666lcz — APPROVED** [review]
Great work!

**Pull request merged**